File: dia-data-import/src/main/java/ie/ncirl/diaproject/dataimport/measurement/tcpthroughput/DeviceProperties.java
package ie.ncirl.diaproject.dataimport.measurement.tcpthroughput;

import ie.ncirl.diaproject.dataimport.measurement.Measurement;

public class DeviceProperties extends Measurement {
    public String battery_level;
    public String cell_info;
    public String timestamp;
    public String network_type;
    public String os_version;
    public DeviceInfo device_info;
    public String carrier;
    public Location location;
    public String rssi;
    public String app_version;
    public String location_type;
    public String is_battery_charging;
    public String[] host_apps;
    public String registration_id;
    public String mobilyzer_version;
    public String request_app;
    public String ssid;
    public String bssid;
    public String cell_rssi;
    public String country_code;
    public String wifi_ip_address;

    @Override
    public String toHdr(String sep) throws UnsupportedOperationException {
        return null;
    }

    @Override
    public String toCsv(String nullValue, String quote, String sep) throws UnsupportedOperationException {
        return null;
    }

    @Override
    public String toNullCsv(String nullValue, String sep) throws UnsupportedOperationException {
        return null;
    }
}

File: ssp-spark-app/src/main/scala/ie/ncirl/sspproject/dataprocess/measurement/DeviceProperties.scala
package ie.ncirl.sspproject.dataprocess.measurement

case class DeviceProperties
(
  battery_level: Integer,
  cell_info: String,
  timestamp: String,
  network_type: String,
  os_version: String,
  device_info: DeviceInfo,
  carrier: String,
  location: Location,
  rssi: Integer,
  app_version: String,
  location_type: String,
  is_battery_charging: String,
  host_apps: Array[String],
  registration_id: String,
  mobilyzer_version: String,
  request_app: String,
  ssid: String,
  bssid: String,
  cell_rssi: String,
  wifi_ip_address: String,
  country_code: String
)

case class DeviceInfo
(
  model: String,
  os: String,
  manufacturer: String,
  tac: String,
  id: String,
  user: String
)

case class Location
(
  latitude: String,
  longitude: String
)
